frontend/ui/src/app/services/cache.ts

```diff
@@ -1,46 +1,71 @@
-import {combineLatest, map, Observable, scan, shareReplay, startWith, Subject} from 'rxjs';
+import {concat, map, Observable, scan, shareReplay, Subject} from 'rxjs';
 import {compareBy, distinctBy, Predicate} from '../../util/arrays';
 import {BaseModel, Named} from '@glasskube/distr-sdk';
 
+interface ReactiveListEvent<T> {
+  objects?: T[]; // used for type 'reset'
+  object?: T; // used when type is 'save' or 'remove'
+  type: ReactiveListEventType;
+}
+
+type ReactiveListEventType = 'save' | 'remove' | 'reset';
```

Comment:
I think this could be improved slightly using a union type, something like this (I didn't try it):
```ts
type ReactiveListEvent<T> = 
  | {type: 'save' | 'remove', object: T}
  | {type: 'reset', objects: T[]};
```

This way, we could rely on the type system to ensure that object/objects is set appropriately and not undefined where it shouldn't be, simplifying the if expression below.